src/Service.Tests/Unittests/SqlMetadataProviderUnitTests.cs

```diff
@@ -339,5 +340,121 @@ public void ValidateGraphQLReservedNaming_DatabaseColumns(string dbColumnName, s
                 actual: isViolationWithGraphQLGloballyDisabled,
                 message: "Unexpected failure. fieldName: " + dbColumnName + " | fieldMapping:" + mappedName);
         }
+
+        /// <summary>
+        /// Test to validate successful inference of relationship data based on data provided in the config and the metadata
+        /// collected from the MsSql database.
+        /// </summary>
+        [TestMethod, TestCategory(TestCategory.MSSQL)]
+        public async Task ValidateInferredRelationshipInfoForMsSql()
+        {
+            DatabaseEngine = TestCategory.MSSQL;
+            await SetupTestFixtureAndInferMetadata();
+            ValidateInferredRelationshipInfoForTables();
+        }
+
+        /// <summary>
+        /// Test to validate successful inference of relationship data based on data provided in the config and the metadata
+        /// collected from the MySql database.
+        /// </summary>
+        [TestMethod, TestCategory(TestCategory.MYSQL)]
+        public async Task ValidateInferredRelationshipInfoForMySql()
+        {
+            DatabaseEngine = TestCategory.MYSQL;
+            await SetupTestFixtureAndInferMetadata();
+            ValidateInferredRelationshipInfoForTables();
+        }
+
+        /// <summary>
+        /// Test to validate successful inference of relationship data based on data provided in the config and the metadata
+        /// collected from the PgSql database.
+        /// </summary>
+        [TestMethod, TestCategory(TestCategory.POSTGRESQL)]
+        public async Task ValidateInferredRelationshipInfoForPgSql()
+        {
+            DatabaseEngine = TestCategory.POSTGRESQL;
+            await SetupTestFixtureAndInferMetadata();
+            ValidateInferredRelationshipInfoForTables();
+        }
+
+        /// <summary>
+        /// Helper method for tests which validate that the relationship data is correctly inferred based on the info provided
+        /// in the config and the metadata collected from the database. It runs the test against various test cases verifying that
+        /// when a relationship is defined in the config between source and target entity and an FK constraint exists in the database between the two entities: We successfully determine which is the referencing
+        /// entity based on the FK constraint.
+        /// </summary>
+        private static void ValidateInferredRelationshipInfoForTables()
```

Comment:
do we have test coverage to handle this scenario?

// Scenario 2: When no FK constraint is defined between source and target entities, but the relationship fields are configured through config file
                        // In this case, two entries will be created. 
                        // First entry: Referencing table: Source entity, Referenced table: Target entity
                        // Second entry: Referencing table: Target entity, Referenced table: Source entity

Reply:
Please refer: https://github.com/Azure/data-api-builder/pull/2051#discussion_r1533581943